tests/destination_create_mock_test.go

```diff
@@ -0,0 +1,299 @@
+		HTTP_PATH,
+		MASKED, // parsonal_access_token
+		CREATE_EXTERNAL_TABLES,
+		EXTERNAL_LOCATION,
+		AUTH_TYPE,
+		ROLE_ARN,
+		MASKED, // secret_key
+		MASKED, // private_key
+		PUBLIC_KEY,
+		CLUSTER_ID,
+		CLUSTER_REGION,
+	)
+}
+
+func prepareConfig() *fivetran.DestinationConfig {
+	config := fivetran.NewDestinationConfig()
+	config.Host(HOST)
+	config.Port(PORT)
+	config.Database(DATABASE)
+	config.Auth(AUTH)
+	config.User(USER)
+	config.Password(PASSWORD)
+	config.ConnectionType(CONNECTION_TYPE)
+	config.TunnelHost(TUNNEL_HOST)
+	config.TunnelPort(TUNNEL_PORT)
+	config.TunnelUser(TUNNEL_USER)
+	config.ProjectID(PROJECT_ID)
+	config.Bucket(BUCKET)
+	config.ServerHostName(SERVER_HOST_NAME)
+	config.HTTPPath(HTTP_PATH)
+	config.PersonalAccessToken(PERSONAL_ACCESS_TOKEN)
+	config.CreateExternalTables(CREATE_EXTERNAL_TABLES)
+	config.ExternalLocation(EXTERNAL_LOCATION)
+	config.AuthType(AUTH_TYPE)
+	config.RoleArn(ROLE_ARN)
+	config.SecretKey(SECRET_KEY)
+	config.PrivateKey(PRIVATE_KEY)
+	config.ClusterId(CLUSTER_ID)
+	config.ClusterRegion(CLUSTER_REGION)
+	return config
+}
+
+func assertRequest(t *testing.T, request map[string]interface{}) {
+	assertKey(t, "service", request, SERVICE)
+	assertKey(t, "region", request, REGION)
+	assertKey(t, "time_zone_offset", request, TIME_ZONE)
+	assertKey(t, "trust_certificates", request, TRUST_CERTIFICATES)
+	assertKey(t, "trust_fingerprints", request, TRUST_FINGERPRINTS)
+	assertKey(t, "run_setup_tests", request, RUN_SETUP_TESTS)
+	assertKey(t, "group_id", request, ID)
+
+	c, ok := request["config"]
+	assertEqual(t, ok, true)
+	config, ok := c.(map[string]interface{})
+	assertEqual(t, ok, true)
+
+	assertKey(t, "host", config, HOST)
+	assertKey(t, "port", config, float64(PORT)) // json marshalling stores all numbers as float64
+	assertKey(t, "database", config, DATABASE)
+	assertKey(t, "auth", config, AUTH)
+	assertKey(t, "user", config, USER)
+	assertKey(t, "password", config, PASSWORD)
+	assertKey(t, "connection_type", config, CONNECTION_TYPE)
+	assertKey(t, "tunnel_host", config, TUNNEL_HOST)
+	assertKey(t, "tunnel_port", config, TUNNEL_PORT)
+	assertKey(t, "tunnel_user", config, TUNNEL_USER)
+	assertKey(t, "project_id", config, PROJECT_ID)
+	assertKey(t, "bucket", config, BUCKET)
+	assertKey(t, "server_host_name", config, SERVER_HOST_NAME)
+	assertKey(t, "http_path", config, HTTP_PATH)
+	assertKey(t, "personal_access_token", config, PERSONAL_ACCESS_TOKEN)
+	assertKey(t, "create_external_tables", config, CREATE_EXTERNAL_TABLES)
+	assertKey(t, "external_location", config, EXTERNAL_LOCATION)
+	assertKey(t, "auth_type", config, AUTH_TYPE)
+	assertKey(t, "role_arn", config, ROLE_ARN)
+	assertKey(t, "secret_key", config, SECRET_KEY)
+	assertKey(t, "private_key", config, PRIVATE_KEY)
+	assertKey(t, "cluster_id", config, CLUSTER_ID)
+	assertKey(t, "cluster_region", config, CLUSTER_REGION)
+}
+
+func assertKey(t *testing.T, key string, requestPart map[string]interface{}, expectedValue interface{}) {
```

Comment:
We should move the `assertKey` function to the `fivetran_tests_common.go` helper file.